app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -89,6 +93,34 @@ class SparkRestClient(sparkConf: SparkConf) {
     }
   }
 
+  def fetchLogData(appId: String, attemptId: Option[String])(
```

Comment:
This is a very good point. I've considered extending `SparkRestDerivedData`, but decided against it, because having two places for the same thing is rarely a good idea. 

Anyway, I've tried implementing it and it doesn't look bad:

* `SparkRestDerivedData.logDerivedData` is now package-private and therefore cannot escape `spark`.
* `restEventLogData` was renamed to `getLogData` (not `getEventLogData`, because the rest of the code does not mention events) and stripped down: no more name checking logic and appId/attempId in the signature.